### PR TITLE
 #377 Extract SchemaTable component

### PR DESCRIPTION
**Pull request by GeorgiChochov**

### CODE REVIEW

File: menas/ui/components/SchemaTable.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+class SchemaTable {
+
+  constructor(oController) {
+    this._oController = oController;
+    this._model;
```
Comment (Zejnilovic) [inline menas/ui/components/SchemaTable.js]: Is this intentional?

File: menas/ui/components/SchemaTable.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    this._schemaTable = oController.byId("schemaFieldsTreeTable")
```
Comment (Zejnilovic) [inline menas/ui/components/SchemaTable.js]: missing `;`

File: menas/ui/components/SchemaTable.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+  get oController() {
+    return this._oController;
+  }
+
+  get model() {
+    return this._model
+  }
+
+  set model(newModel) {
+    newModel = new sap.ui.model.json.JSONModel(newModel)
```
Comment (Zejnilovic) [inline menas/ui/components/SchemaTable.js]: missing `;`

File: menas/ui/components/SchemaTable.js [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    this.schemaTable.setModel(newModel, "schema");
+    this._model = newModel
+  }
+
+  metadataPress(oEv) {
+    let binding = oEv.getSource().getBindingContext("schema").getPath() + "/metadata";
+    let bindingArr = binding + "Arr";
+    //hmm bindAggregation doesn't take formatter :-/
+    const model = this.oController.byId("schemaFieldsTreeTable").getModel("schema")
```
Comment (Zejnilovic) [inline menas/ui/components/SchemaTable.js]: missing `;`

File: menas/ui/components/SchemaTable.js (hunk 1 above)
Comment (GeorgiChochov) [reply menas/ui/components/SchemaTable.js]: It was, but I guess I don't need to declare the field in the constructor if I leave it undefined.

File: menas/ui/components/SchemaTable.js [hunk 5]
```diff
@@ -31,11 +30,11 @@ class SchemaTable {
   }
 
   get model() {
-    return this._model
+    return this._model;
```
Comment (Zejnilovic) [inline menas/ui/components/SchemaTable.js]: I think we don't need a semicolon here. As to why I refer to [EC6 Language Specification](https://www.ecma-international.org/ecma-262/7.0/index.html#sec-automatic-semicolon-insertion)

Comment (GeorgiChochov) [reply menas/ui/components/SchemaTable.js]: Good point. (I'm actually leaning towards forgoing the semicolons entirely)

**Comment by Zejnilovic** [comment]
There is an issue introduced by this PR it seems (as in I cannot reproduce it on develop)
When I try to change the schema in new/edit pop-up, the versions don't update.

May-10-2019 18-23-04

File: menas/ui/components/SchemaTable.js (hunk 5 above)
Comment (Zejnilovic) [reply menas/ui/components/SchemaTable.js]: I like them to be honest. But I have no real reason to be against them. It is just a readability thing for me. It eases my mind when I am sure that the line ends there.

**Comment by GeorgiChochov** [comment]
> There is an issue introduced by this PR it seems (as in I cannot reproduce it on develop)
> When I try to change the schema in new/edit pop-up, the versions don't update.
> 
> May-10-2019 18-23-04

Thanks for spotting it!

**Comment by Zejnilovic** [comment]
Second Issue. When I update a dataset's schema while having the schema tab open, it does not update.

**Comment by Zejnilovic** [comment]
First issue solved.

**Comment by Zejnilovic** [comment]
There still seems to be a problem with schemas view as to it not updating. After refresh I had version 7.

May-13-2019 16-08-23

**Comment by Zejnilovic** [comment]
Now I am not sure if this is a feature or another issue, but when you update any model, the left menu with others and their latest versions doesn't update

I am talking about this left menu.
image

**Pull request merged**